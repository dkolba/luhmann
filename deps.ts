export { render } from "https://deno.land/x/gfm@0.2.2/mod.ts";
export { parse } from "https://deno.land/std@0.186.0/yaml/mod.ts";
export {
  extract,
  test,
} from "https://deno.land/std@0.186.0/front_matter/yaml.ts";
export * as log from "https://deno.land/std@0.186.0/log/mod.ts";
