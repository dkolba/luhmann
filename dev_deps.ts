export { assertSnapshot } from "https://deno.land/std@0.188.0/testing/snapshot.ts";
