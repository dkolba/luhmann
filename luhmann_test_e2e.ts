import { assert, assertSnapshot } from "./dev_deps.ts";
import { parse } from "./deps.ts";

const baseFileUrl = " http://localhost:4507/";
const baseTestUrl = "http://localhost:8080/";
const sitemapResource = "sitemap.yaml";

Deno.test("e2e isEtagHit", async function (): Promise<void> {
  // Get the homepage the first time
  const homePageResponse = await fetch(baseTestUrl);
  // Dispose of the response to prevent a 'leaking resources' error
  await homePageResponse.body?.cancel();
  // Extract the etag from header for reuse in the next request
  const etag = await homePageResponse.headers.get("etag");
  const headers = { headers: { "If-None-Match": etag ? etag : "NOETAG" } };
  // Get the homepage the second time, now with If-None-Match header
  const homePageIfNoneMatch = await fetch(baseTestUrl, headers);
  // Again, dispose of the response to prevent a 'leaking resources' error
  await homePageIfNoneMatch.body?.cancel();

  await assert(homePageIfNoneMatch.status === 304);
});

Deno.test("e2e isSnapshotMatch", async function (t): Promise<void> {
  const homePageResponse = await fetch(baseTestUrl);
  const homePageText = await homePageResponse.text();

  await assertSnapshot(t, homePageText);

  const sitemapResponse = await fetch(`${baseFileUrl}${sitemapResource}`);
  const sitemapbody = await sitemapResponse?.text();
  const parsedYamlDocList = parse(sitemapbody) as string[];

  for (const doc of parsedYamlDocList) {
    const mdName = doc.split(".").slice(0, -1).join(".");
    const docResponse = await fetch(`${baseTestUrl}${mdName}`);
    const docText = await docResponse.text();

    await assertSnapshot(t, docText);
  }
});
