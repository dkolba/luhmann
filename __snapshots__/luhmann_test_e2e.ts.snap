export const snapshot = {};

snapshot[`e2e isSnapshotMatch 1`] = `
'<!doctype html>
  <html lang=en>
    <head>
      <meta charset=utf-8>
      <title>Zettelkasten</title>
      <link rel="stylesheet" href="http://localhost:4507/paper.css" />
      <style>
        main {
          max-width: 800px;
          margin: 0 auto;
        }
        
      </style>
      <script type="module" src="http://localhost:4507/welcome.js"></script>
    </head>
    <body>
      <main data-color-mode="light" data-light-theme="light" data-dark-theme="dark" class="markdown-body">
      <section>
     <h2><a href="/test_zettel_1">A first test title</a></h2>
     <time datetime="Fri, 21 May 2032 00:00:00 GMT">21.5.2032</time>
     <span>Best test teaser in the industry</span>
     <span> - </span><time datetime="Fri, 21 May 2032 00:00:00 GMT">2032/4/5</time>
\\t</section><section>
     <h2><a href="/test_zettel_2">A second test title</a></h2>
     <time datetime="Mon, 24 Dec 2001 00:00:00 GMT">24.12.2001</time>
     <span>Second best test teaser in the industry</span>
     <span> - </span><time datetime="Mon, 24 Dec 2001 00:00:00 GMT">2001/11/1</time>
\\t</section>
    </main>
  </body>
</html>
'
`;

snapshot[`e2e isSnapshotMatch 2`] = `
'<!doctype html>
  <html lang=en>
    <head>
      <meta charset=utf-8>
      <title>Zettelkasten</title>
      <link rel="stylesheet" href="http://localhost:4507/paper.css" />
      <style>
        main {
          max-width: 800px;
          margin: 0 auto;
        }
        
      </style>
      <script type="module" src="http://localhost:4507/welcome.js"></script>
    </head>
    <body>
      <main data-color-mode="light" data-light-theme="light" data-dark-theme="dark" class="markdown-body">
      <section>
     <h2><a href="/test_zettel_1">A first test title</a></h2>
     <time datetime="Fri, 21 May 2032 00:00:00 GMT">21.5.2032</time>
     <span>Best test teaser in the industry</span>
     <span> - </span><time datetime="Fri, 21 May 2032 00:00:00 GMT">2032/4/5</time>
\\t</section><h2 id="test-headline-1"><a class="anchor" aria-hidden="true" tabindex="-1" href="#test-headline-1"><svg class="octicon octicon-link" viewbox="0 0 16 16" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M7.775 3.275a.75.75 0 001.06 1.06l1.25-1.25a2 2 0 112.83 2.83l-2.5 2.5a2 2 0 01-2.83 0 .75.75 0 00-1.06 1.06 3.5 3.5 0 004.95 0l2.5-2.5a3.5 3.5 0 00-4.95-4.95l-1.25 1.25zm-4.69 9.64a2 2 0 010-2.83l2.5-2.5a2 2 0 012.83 0 .75.75 0 001.06-1.06 3.5 3.5 0 00-4.95 0l-2.5 2.5a3.5 3.5 0 004.95 4.95l1.25-1.25a.75.75 0 00-1.06-1.06l-1.25 1.25a2 2 0 01-2.83 0z"></path></svg></a>Test Headline 1</h2><p>An unordered test list</p>
<ul>
<li>Test list item 1</li>
<li>Test list item 2</li>
<li>Test list item 3</li>
</ul>
<p>An ordered test list</p>
<ol>
<li>Test list item 4</li>
<li>Test list item 5</li>
<li>Test list item 6</li>
</ol>
<h3 id="a-test-subheadling"><a class="anchor" aria-hidden="true" tabindex="-1" href="#a-test-subheadling"><svg class="octicon octicon-link" viewbox="0 0 16 16" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M7.775 3.275a.75.75 0 001.06 1.06l1.25-1.25a2 2 0 112.83 2.83l-2.5 2.5a2 2 0 01-2.83 0 .75.75 0 00-1.06 1.06 3.5 3.5 0 004.95 0l2.5-2.5a3.5 3.5 0 00-4.95-4.95l-1.25 1.25zm-4.69 9.64a2 2 0 010-2.83l2.5-2.5a2 2 0 012.83 0 .75.75 0 001.06-1.06 3.5 3.5 0 00-4.95 0l-2.5 2.5a3.5 3.5 0 004.95 4.95l1.25-1.25a.75.75 0 00-1.06-1.06l-1.25 1.25a2 2 0 01-2.83 0z"></path></svg></a>A test subheadling</h3><p>Some test text</p>

    </main>
  </body>
</html>
'
`;

snapshot[`e2e isSnapshotMatch 3`] = `
'<!doctype html>
  <html lang=en>
    <head>
      <meta charset=utf-8>
      <title>Zettelkasten</title>
      <link rel="stylesheet" href="http://localhost:4507/paper.css" />
      <style>
        main {
          max-width: 800px;
          margin: 0 auto;
        }
        
      </style>
      <script type="module" src="http://localhost:4507/welcome.js"></script>
    </head>
    <body>
      <main data-color-mode="light" data-light-theme="light" data-dark-theme="dark" class="markdown-body">
      <section>
     <h2><a href="/test_zettel_2">A second test title</a></h2>
     <time datetime="Mon, 24 Dec 2001 00:00:00 GMT">24.12.2001</time>
     <span>Second best test teaser in the industry</span>
     <span> - </span><time datetime="Mon, 24 Dec 2001 00:00:00 GMT">2001/11/1</time>
\\t</section><h2 id="test-headline-2"><a class="anchor" aria-hidden="true" tabindex="-1" href="#test-headline-2"><svg class="octicon octicon-link" viewbox="0 0 16 16" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M7.775 3.275a.75.75 0 001.06 1.06l1.25-1.25a2 2 0 112.83 2.83l-2.5 2.5a2 2 0 01-2.83 0 .75.75 0 00-1.06 1.06 3.5 3.5 0 004.95 0l2.5-2.5a3.5 3.5 0 00-4.95-4.95l-1.25 1.25zm-4.69 9.64a2 2 0 010-2.83l2.5-2.5a2 2 0 012.83 0 .75.75 0 001.06-1.06 3.5 3.5 0 00-4.95 0l-2.5 2.5a3.5 3.5 0 004.95 4.95l1.25-1.25a.75.75 0 00-1.06-1.06l-1.25 1.25a2 2 0 01-2.83 0z"></path></svg></a>Test Headline 2</h2><p>An unordered test list</p>
<ul>
<li>Test list item 1</li>
<li>Test list item 2</li>
<li>Test list item 3</li>
</ul>
<p>An ordered test list</p>
<ol>
<li>Test list item 4</li>
<li>Test list item 5</li>
<li>Test list item 6</li>
</ol>
<h3 id="a-test-subheadling"><a class="anchor" aria-hidden="true" tabindex="-1" href="#a-test-subheadling"><svg class="octicon octicon-link" viewbox="0 0 16 16" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M7.775 3.275a.75.75 0 001.06 1.06l1.25-1.25a2 2 0 112.83 2.83l-2.5 2.5a2 2 0 01-2.83 0 .75.75 0 00-1.06 1.06 3.5 3.5 0 004.95 0l2.5-2.5a3.5 3.5 0 00-4.95-4.95l-1.25 1.25zm-4.69 9.64a2 2 0 010-2.83l2.5-2.5a2 2 0 012.83 0 .75.75 0 001.06-1.06 3.5 3.5 0 00-4.95 0l-2.5 2.5a3.5 3.5 0 004.95 4.95l1.25-1.25a.75.75 0 00-1.06-1.06l-1.25 1.25a2 2 0 01-2.83 0z"></path></svg></a>A test subheadling</h3><p>Some test text</p>

    </main>
  </body>
</html>
'
`;
